TradeEngine/src/main/java/com/gala/sam/tradeEngine/service/MarketService.java

```diff
@@ -48,63 +61,115 @@ void init() {
+      return Optional.empty();
+    }
+
+    try {
+      processTriggeredStopOrders();
+    } catch (ProcessingStopOrderException e) {
+      log.error("Not all stop orders have been handled correctly since an exception was raised in processing them: {}", e.getStackTrace());
+    }
+
+    return Optional.of(order);
   }
 
   public List<Trade> getAllMatchedTrades() {
     return marketState.getTrades();
   }
 
-  private void processOrder(Order order) {
+  private void processOrder(AbstractOrder order)
+      throws OrderTypeNotSupportedException, OrderTimeInForceNotSupportedException, OrderDirectionNotSupportedException {
     log.info(String.format("Processing order %s", order.toString()));
 
-    orderProcessorFactory.getOrderProcessor(marketState, order.getType())
-        .process(order);
+    AbstractOrderProcessor orderProcessor = orderProcessorFactory
+        .getOrderProcessor(marketState, order.getType());
 
-    log.info("Ticker queues: " + marketState.getTickerQueues().toString());
-    log.info("Stop Orders: " + marketState.getStopOrders().toString());
-    log.info("Trades: " + marketState.getTrades().toString());
+    handleOrderWithTimer(order, orderProcessor);
+
+    log.debug("Ticker queues: " + marketState.getTickerQueues().toString());
+    log.debug("Stop Orders: " + marketState.getStopOrders().toString());
+    log.debug("Trades: " + marketState.getTrades().toString());
+  }
+
+  private void handleOrderWithTimer(AbstractOrder order, AbstractOrderProcessor orderProcessor)
+      throws OrderTimeInForceNotSupportedException, OrderDirectionNotSupportedException {
+    StopWatch orderProcessorTimer = new StopWatch();
+    orderProcessorTimer.start();
+    orderProcessor.process(order);
+    orderProcessorTimer.stop();
+    log.info("Order {} was handled in {} milliseconds", order.getOrderId(),
+        orderProcessorTimer.getTotalTimeMillis());
   }
 
-  private void processTriggeredStopOrders() {
-    Iterator<StopOrder> it = marketState.getStopOrders().iterator();
+  private void processTriggeredStopOrders() throws ProcessingStopOrderException {
+    Iterator<AbstractStopOrder> it = marketState.getStopOrders().iterator();
     while (it.hasNext()) {
-      StopOrder stopOrder = it.next();
-      log.info("Testing Trigger on: " + stopOrder.toString());
+      AbstractStopOrder stopOrder = it.next();
+      log.debug("Testing Trigger on: " + stopOrder.toString());
       if (isStopLossTriggered(stopOrder)) {
-        log.info("Stop OrderReq Triggered");
+        log.debug("Stop order request Triggered");
         it.remove();
         orderRepository.delete(stopOrder);
-        ActiveOrder activeOrder = stopOrder.toActiveOrder();
-        processOrder(activeOrder);
+        AbstractActiveOrder activeOrder = stopOrder.toActiveOrder();
+        try {
+          processOrder(activeOrder);
+        } catch (AbstractOrderFieldNotSupportedException e) {
+          throw new ProcessingStopOrderException(stopOrder, e);
```

Comment:
Replace exception with logging.

If you're processing a collection of orders, and there's a problem with one of them, do you want to stop processing completely, or just skip that one and process the rest of them? In this case, I think that you want to do the latter. However, if you throw and exception here, you'll do the former.